tests/FunctionalResolverTest.php

```diff
@@ -75,6 +75,46 @@ public function testResolveGoogleOverTcpResolves()
         $this->loop->run();
     }
 
+    /**
+     * @group internet
+     */
+    public function testResolveGoogleOverTlsResolves()
+    {
+        if (defined('HHVM_VERSION')) {
+            $this->markTestSkipped('Not supported on legacy HHVM');
+        } elseif (\PHP_VERSION_ID < 50600) {
+            $this->markTestSkipped('Not supported on legacy PHP < 5.6');
+        }
+
+        $factory = new Factory();
+        $this->resolver = $factory->create('tls://8.8.8.8', $this->loop);
+
+        $promise = $this->resolver->resolve('google.com');
+        $promise->then($this->expectCallableOnce(), $this->expectCallableNever());
+
+        $this->loop->run();
+    }
+
+    /**
+     * @group internet
+     */
+    public function testAttemptTlsOnNonTlsPortRejects()
+    {
+        if (defined('HHVM_VERSION')) {
+            $this->markTestSkipped('Not supported on legacy HHVM');
+        } elseif (\PHP_VERSION_ID < 50600) {
+            $this->markTestSkipped('Not supported on legacy PHP < 5.6');
+        }
```

Comment:
Would it make sense to do something like this here? (And in the test above as well.)

```suggestion
        if (defined('HHVM_VERSION') && \PHP_VERSION_ID < 50600) {
            self::expectedException(\BadMethodCallException);
        }
```

Reply:
The exception is wrapped in the promise reject, so setExpectedException didn't handle the reject as expected. Instead I've added an additional test case specifically for legacy PHP versions to check the rejected promise contains a BadMethodCallException